plugins/pdr_deterministic_plugin/ufm_sim_web_service/isolation_mgr.py

```diff
@@ -97,6 +119,36 @@ def eval_isolation(self, port_name, cause):
         if not self.test_mode:
             self.ufm_client.send_event(log_message, event_id=Constants.EXTERNAL_EVENT_ALERT, external_event_name="Isolating Port")
 
+    def reset_port(self, port_name, port_guid):
+        """
+        Reset port if reset limit is not not exceeded
+        """
+        # Check if reset is allowed
+        reset_history = self.ports_resets.get(port_name)
+        if reset_history:
+            if datetime.now() - reset_history.reset_time > self.port_reset_interval:
+                # Passed too much time from last reset: clean reset history and allow resets
+                del self.ports_resets[port_name]
+            elif reset_history.reset_count >= self.max_port_reset_num:
+                # Exceeds reset limit
+                self.logger.info("Skipping reset of port: %s... reset limit exceeded", port_name)
+                return
+
+        # Perform reset
+        ret = self.ufm_client.reset_port(port_name, port_guid)
+        if not ret or ret.status_code != http.HTTPStatus.OK:
+            self.logger.warning("Failed resetting port: %s... status_code= %s", port_name, ret.status_code)
+            return
+
+        # Update port resets history
+        reset_history = self.ports_resets.get(port_name)
+        if not reset_history:
+            reset_history = PortReset(port_name)
+            self.ports_resets[port_name] = reset_history
+
+        reset_history.reset_count += 1
+        reset_history.reset_time = datetime.now()
```

Comment:
because you put the reset time here, which is excellent, you can use it to reset the restarts counter if enough time has passed, but it is only something to add in the future.

Reply:
The reset the restarts counter is performed during reset conditions check before the actual reset